ProjFS.Mac/PrjFSKextTests/MockVnodeAndMount.hpp

```diff
@@ -16,11 +25,17 @@ struct mount
 private:
     vfsstatfs statfs;
     uint64_t nextInode;
-    
+    std::weak_ptr<mount> weakSelfPointer;
+    std::weak_ptr<vnode> rootVnode;
```

Comment:
> What's the reason for using a `weak_ptr` for `rootVnode` rather than a `shared_ptr`?

This way, we can have a mock mount point as a test class instance variable and don't have to manually clear out the `rootVnode` after every test.

Generally, the idea is that mock `vnode` objects are all destroyed automatically by the end of each test because their `shared_ptr` references are all: (1) local to the test or (2) transitive (the `parent` reference in `vnode`).

If they aren't, that should indicate a bug, and is immediately caught by the assertion in `MockVnodes_CheckAndClear()`.

Reply:
Thanks all that makes sense!

If you don't mind, I think documenting this could be as simple as added a comment to the top of MockVnodeAndMount.hpp that includes what you've described here.